src/Entity/Transformer.php

```diff
@@ -29,6 +29,9 @@
  * This class transform a SnapshotInterface into PageInterface.
  *
  * @final since sonata-project/page-bundle 3.26
+ *
+ * @phpstan-import-type PageContent from TransformerInterface
+ * @phpstan-import-type BlockContent from TransformerInterface
  */
 class Transformer implements TransformerInterface
```

Comment:
if we going to remove this class on 4.x we should deprecate this class to remove on the next major!

Reply:
i don't plan on removing this class in 4.x, just to use Serializer

Reply:
But here leads to a problem if we deprecate this class to remove on 4.x we are not providing any other alternative for whom is extending this class!

should we implement the serializer solution in 4.x? if yes then we should deprecate this class together with serializer solution!

what do you think @VincentLanglet ?

Reply:
So far the class is not removed and the implemenation of the Serializer is not provided. So that's another topic.

Reply:
Ok!

Reply:
> i don't plan on removing this class in 4.x, just to use Serializer

hmm I thought that, it could be removed! because it is doing the serializer functionality! in theory you just need to pass the object into the normalizer like `$serializer->normalizer($block)`

Reply:
yeah, the Serializer will be called inside the Transformer when doing:
`$transformer->create($page) => $snapshot`

i didn't wanted to do it in this MR yet, because it most likely will be a BC break

Reply:
ok :)